packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -116,6 +127,36 @@ class ChildRange extends QueryPredicateOperator<ChildRange> {
   }
 }
 
+/**
+ * A predicate operator that modifies the range of the match to shrink to regex
+ * match.  For example, `(#shrink-to-match! @foo "\\S+")` will modify the range
+ * of the `@foo` capture to exclude whitespace.
+ */
+class ShrinkToMatch extends QueryPredicateOperator<ShrinkToMatch> {
+  name = "shrink-to-match!" as const;
+  schema = z.tuple([q.node, q.string]);
+
+  run(nodeInfo: MutableQueryCapture, pattern: string) {
+    const {
+      range,
+      node: { text },
+    } = nodeInfo;
+
+    const match = text.match(new RegExp(pattern));
```

Comment:
Is it a problem that we're using the text from the node even though the range might have changed?

Reply:
Huh. Yeah we should prob add editor and get text that way?

Reply:
Does make me a bit nervous that we have two ways to get this stuff that can become out of sync 😬

Reply:
I agree. Maybe we should make a simple node type. It basically only needs its type and its children and parent?

Reply:
Yeah and then I guess it would return the same type from those. I'd start with just those fields

Reply:
Agreed https://github.com/cursorless-dev/cursorless/issues/1725